src/anyp/PortCfg.cc

```diff
@@ -25,13 +28,41 @@ AnyP::PortCfgPointer FtpPortList;
 int NHttpSockets = 0;
 int HttpSockets[MAXTCPLISTENPORTS];
 
-AnyP::PortCfg::PortCfg(const TrafficModeFlags::PortKind aPortKind):
+static AnyP::TrafficModeFlags::PortKind
+PortKind(const SBuf &directive)
+{
+    if (directive.cmp("http_port") == 0)
+        return AnyP::TrafficModeFlags::httpPort;
+    else if (directive.cmp("https_port") == 0)
+        return AnyP::TrafficModeFlags::httpsPort;
+    else {
+        assert(directive.cmp("ftp_port") == 0);
+        return AnyP::TrafficModeFlags::ftpPort;
+    }
+}
+
+static AnyP::ProtocolVersion
+DefaultTransport(const AnyP::TrafficModeFlags::PortKind &portKind)
+{
+    switch(portKind)
+    {
+    case AnyP::TrafficModeFlags::httpPort:
+        return Http::ProtocolVersion(1,1);
+    case AnyP::TrafficModeFlags::httpsPort:
+        return AnyP::ProtocolVersion(AnyP::PROTO_HTTPS, 1,1);
+    case AnyP::TrafficModeFlags::ftpPort:
+        return Ftp::ProtocolVersion();
+    }
+}
+
```

Comment:
Please move both static helpers into the Anyp namespace to get rid of lots of `Anyp::` prefixes. You do not need to forward-declare these functions for that. Just declare/define them inside one `namespace Anyp {...}` scope.